File: 3351-maximize-happiness-of-selected-children/maximize-happiness-of-selected-children.cpp
#define ll long long
class Solution {
public:
    long long maximumHappinessSum(vector<int>& happiness, int k) {
        priority_queue<ll> pq;
        for(auto h: happiness){
            pq.push(h);
        }
        ll i=0;
        ll ans = 0;
        while(pq.size() && k){
            if(pq.top() - i>=0){
                ans +=pq.top() - i;
                i++;k--;
            }
            else break;
            pq.pop();
        }
        return ans;
    }
};

File: 3351-maximize-happiness-of-selected-children/maximize-happiness-of-selected-children.java
class Solution {
    public long maximumHappinessSum(int[] happiness, int k) {
        PriorityQueue<Integer> pq = new PriorityQueue(Collections.reverseOrder());
        for(int h: happiness){
            pq.add(h);
        }
        long ans = 0;
        long i = 0;
        while(!pq.isEmpty() && k>0){
            if(pq.peek()-i>0){
                ans += pq.peek()-i;
                i++;k--;
            }else break;
            pq.poll();
        }
        return ans;
    }
}